### PR TITLE
Feature/members list

### PR DESCRIPTION
**Pull request by skuhlmann**
## GitHub Issue

[issue](https://github.com/HausDAO/daohaus-monorepo/issues/511)

## Changes

- members table
- dao context updates for member sort and pagination
- load more

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

**Review by jordanlesich — APPROVED** [review]
Approved after some small changes.

File: apps/core-app/src/utils/contextHelpers.ts [hunk 1]
```diff
@@ -83,11 +124,10 @@ export const loadMembersList = async ({
     });
 
     if (shouldUpdate) {
-      if (res.nextPaging) {
-        setNextPaging(res.nextPaging);
-      }
+      setNextPaging(res.nextPaging);
 
       setData((prevState) => {
+        if (deepEqual(prevState, res.items)) return res.items;
```
Comment (jordanlesich) [inline apps/core-app/src/utils/contextHelpers.ts:130]: Is this the pattern we're using to deal with the useEffect issue?

**Review by jordanlesich — COMMENTED** [review]
Just want to double check that we're not using the deepEql to deal with the doublefetch issue. Don't think that would really solve it. Let's talk about it on Monday

File: apps/core-app/src/components/DaohausTable.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  tableData: MembersTableType[];
+  columns: Column<MembersTableType>[];
+  hasNextPaging: boolean;
+  handleLoadMore: (event: MouseEvent<HTMLButtonElement>) => void;
+  handleColumnSort: (orderBy: string, orderDirection: 'asc' | 'desc') => void;
+  sortableColumns: string[];
+};
+
+// TS Challenge figure out how to pass generics for the table props
+// type DaoTableProps<T extends object> = {
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply apps/core-app/src/components/DaohausTable.tsx:52]: I got this figured out in my next PR.

DaoTable<T extends object>

File: apps/core-app/src/components/MembersOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { daysToWeeks } from 'date-fns';
+import { formatValueTo, fromWei } from '@daohaus/common-utilities';
+
+const MembersOverviewCard = styled(Card)`
+  background-color: ${({ theme }: { theme: Theme }) => theme.card.hoverBg};
+  border: none;
+  padding: 3rem;
+`;
+
+const DataGrid = styled.div`
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply apps/core-app/src/components/MembersOverview.tsx:14]: I was thinking so - but i'm not css strong enough for it... i am having to adjust a few things to get spacing correct.

I am going to file a card for it for the future.

File: apps/core-app/src/utils/contextHelpers.ts (hunk 1 above)
Comment (skuhlmann) [reply apps/core-app/src/utils/contextHelpers.ts:130]: > Is this the pattern we're using to deal with the useEffect issue?

Not really - just preventing updating state again. I can't figure out the double use effect firing issue in development/strict mode

**Review by brossetti1 — APPROVED** [review]
this looks really clean to me, not much to add here -- still getting used to the ui lib though

File: apps/core-app/src/contexts/DaoContext.tsx [hunk 4]
```diff
@@ -22,19 +24,30 @@ import {
...
   dao: null,
   isDaoLoading: false,
   refreshDao: async () => {
     return;
   },
+  userMembership: null,
+  isUserMembershipLoading: false,
+  refreshUserMembership: async () => {
+    return;
+  },
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply apps/core-app/src/contexts/DaoContext.tsx:46]: oh this is the default data for the type when setting up the context - the real function is defined below.

**Pull request merged**